### ISSUE TITLE
Allow refined methods to add optional parameters

### ISSUE THREAD

**Issue by gdejohn** [ceylon/ceylon-spec#1230]
```
interface Foo {
	shared formal String foo(String s);
}

abstract class Bar() satisfies Foo {
	shared actual formal String foo(String s, Integer i = 0);
}
```

**Comment by gavinking** [comment]
I think this would be useful and pretty easy to implement.

**Comment by FroMage** [comment]
No, not easy no, there's a lot of logic that relies on that current limitation. I _think_ we didn't add this limitation randomly though and there was a reason for it at the time, though I can't recall it. Perhaps @tombentley remembers?

**Comment by jvasileff** [comment]
There may be an api evolution/compatibility angle to consider, since if `BarSub.foo` in another module previously added some other set of optional parameters, the change to `Bar.foo` would be breaking for `BarSub`.

In theory, it seems like adding parameters without breaking compatibility could be allowed *either* in a) the type at the top that originally declares the member (non-`actual`), or b) a type at the bottom (non-`default` and non-`formal`). But allowing both (or allowing additions in the middle, i.e. `actual default` or `actual formal` refinements) may unexpectedly break things.

**Comment by gavinking** [comment]
@jvasileff but surely adding parameters (even defaulted parameters) to a non-`default` method is *already* a breaking change, no?

**Comment by jvasileff** [comment]
Oops. Yes, of course it’s unavoidable that adding defaulted parameters is a breaking change.
